pkg/porter/explain.go

```diff
@@ -0,0 +1,291 @@
+	}
+
+	err = p.applyDefaultOptions(&o.sharedOptions)
+	if err != nil {
+		return err
+	}
+	err = p.ensureLocalBundleIsUpToDate(o.bundleFileOptions)
+	if err != nil {
+		return err
+	}
+	bundle, err := p.CNAB.LoadBundle(o.CNABFile, o.Insecure)
+	// Print Bundle Details
+
+	pb, err := generatePrintable(bundle)
+	if err != nil {
+		return errors.Wrap(err, "unable to print bundle")
+	}
+	switch o.Format {
+	case printer.FormatJson:
+		return printer.PrintJson(p.Out, pb)
+	case printer.FormatYaml:
+		return printer.PrintYaml(p.Out, pb)
+	case printer.FormatTable:
+		return p.printTable(pb)
+	default:
+		return fmt.Errorf("invalid format: %s", o.Format)
+	}
+}
+
+func generatePrintable(bun *bundle.Bundle) (*PrintableBundle, error) {
+	if bun == nil {
+		return nil, fmt.Errorf("expected a bundle")
+	}
+	pb := PrintableBundle{
+		Name:        bun.Name,
+		Description: bun.Description,
+		Version:     bun.Version,
+	}
+
+	actions := map[string]PrintableAction{}
+	for a, v := range bun.Actions {
+		pa := PrintableAction{}
+		pa.Description = v.Description
+		pa.Modifies = v.Modifies
+		pa.Stateless = v.Stateless
+		actions[a] = pa
+	}
+
+	creds := map[string]PrintableCredential{}
+	for c, v := range bun.Credentials {
+		pc := PrintableCredential{}
+		pc.Description = v.Description
+		pc.Required = v.Required
+
+		creds[c] = pc
+	}
+
+	params := map[string]PrintableParameter{}
+	for p, v := range bun.Parameters {
+		def, ok := bun.Definitions[v.Definition]
+		if !ok {
+			return nil, fmt.Errorf("unable to find definition %s", v.Definition)
+		}
+		if def == nil {
+			return nil, fmt.Errorf("empty definition for %s", v.Definition)
+		}
+		pp := PrintableParameter{}
+		pp.Type = def.Type
+		pp.Default = def.Default
+		pp.ApplyTo = generateApplyToString(v.ApplyTo)
+		pp.Required = v.Required
+		pp.Description = v.Description
+
+		params[p] = pp
+	}
+
+	outputs := map[string]PrintableOutput{}
+	for p, v := range bun.Outputs {
+		def, ok := bun.Definitions[v.Definition]
+		if !ok {
+			return nil, fmt.Errorf("unable to find definition %s", v.Definition)
+		}
+		if def == nil {
+			return nil, fmt.Errorf("empty definition for %s", v.Definition)
+		}
+		pd := PrintableOutput{}
+		pd.Type = def.Type
+		pd.ApplyTo = generateApplyToString(v.ApplyTo)
+		pd.Description = v.Description
+
+		outputs[p] = pd
+	}
+
+	pb.Actions = actions
+	pb.Credentials = creds
+	pb.Outputs = outputs
+	pb.Parameters = params
+
+	return &pb, nil
+}
+
+func (p *Porter) printTable(bun *PrintableBundle) error {
+	fmt.Fprintf(p.Out, "Name: %s\n", bun.Name)
+	fmt.Fprintf(p.Out, "Description: %s\n", bun.Description)
+	fmt.Fprintf(p.Out, "Version: %s\n", bun.Version)
+	fmt.Fprintln(p.Out, "")
+
+	p.generateCredentialsBlock(bun)
+
+	p.generateParametersBlock(bun)
+
+	p.generateOutputsBlock(bun)
+
+	p.generateActionsBlock(bun)
+
+	return nil
+}
+
+func (p *Porter) generateCredentialsBlock(bun *PrintableBundle) {
```

Comment:
nit: These functions below should all be named print* since they do hit stdout and don't return anything

```suggestion
func (p *Porter) printCredentialsBlock(bun *PrintableBundle) {
```